sklearn/model_selection/tests/test_split.py

```diff
@@ -2054,3 +2054,19 @@ def test_no_group_splitters_warns_with_groups(cv):
 
     with pytest.warns(UserWarning, match=msg):
         cv.split(X, y, groups=groups)
+
+
+def test_Stratified_mandatory_y_argument():
+    x = ["a"] * 100
+    # y = [0] * 90 + [1] * 10
+    # Variable 'y' is required for Stratified classes split function.
+    # Testing the split without 'y', shoudl raise TypeError.
+    msg = r"split\(\) missing 1 required positional argument: 'y'"
+
+    cv = StratifiedKFold(n_splits=2)
```

Comment:
Since you are repeating the same code, it is probably useful to take advantage of pytest parametrized tests. You can also add `StratifiedShuffleSplit` to the list of tested classes, so something like this?

```py
splitters_requiring_y = [
    StratifiedKFold(),
    StratifiedShuffleSplit(),
    RepeatedStratifiedKFold(),
]


@pytest.mark.parametrize(
    "cv", splitters_requiring_y, ids=[str(cv) for cv in splitters_requiring_y]
)
def test_stratified_splitter_without_y(cv):
    msg = r"split.+missing 1 required positional argument: 'y'"
    with pytest.raises(TypeError, match=msg):
        cv.split(X)
```

Reply:
I did a new commit with the mentioned changes